bindata/v4.1.0/kube-apiserver/pod.yaml

```diff
@@ -42,6 +42,17 @@ spec:
             echo "Copying system trust bundle"
             cp -f /etc/kubernetes/static-pod-certs/configmaps/trusted-ca-bundle/ca-bundle.crt /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem
           fi
+          echo -n "Waiting for port :6443 and :6080 to be released."
```

Comment:
Look in line 27. Why is that not enough?

Reply:
That handles this case on pod restarts, not container restarts. The InitContainer only runs once.

Reply:
Should we remove it from the init container?

Reply:
I was thinking about that. I decided against it, mostly because the init-container provides a useful bit of feedback w.r.t. pod lifecycle. But I don't hold this opinion particularly strongly.

Reply:
> I was thinking about that. I decided against it, mostly because the init-container provides a useful bit of feedback w.r.t. pod lifecycle. But I don't hold this opinion particularly strongly.

The init container also prevents the time spent waiting from counting against a health check timer.